### PR TITLE
Feat: Improve samples

### PR DESCRIPTION
**Pull request by sbkok**
## Why?

These changes aim to improve the overall quality, maintainability, and
usability of the ADF sample templates, while also providing better
documentation and aligning with the latest AWS best practices.

- To adhere to best practices and improve maintainability of the code base.
- To enhance security and observability by enabling VPC Flow Logs.
- To address code style issues and enforce consistent formatting.

## What?

- Specified explicit files to include in the CodeBuild output artifacts.
- Improvements to the VPC example:
      - Added support for enabling VPC Flow Logs to S3 or CloudWatch Logs.
      - Refactored resource ordering and added conditions for better readability.
      - Exported additional VPC CIDR range output for convenience.
- Fixed minor documentation issues in sample guides.
- Update README files with additional details, prerequisites, and deployment
   instructions for various samples.
- Upgrade the `sample-fargate-node-app` to use the AWS public container registry
   (public.ecr.aws) instead.
- Refactor the `sample-ec2-with-codedeploy` sample to use AWS Launch Templates,
   a newer and recommended approach instead of Launch Configurations.
- Update `sample-ec2-with-codedeploy` scripts to install newer versions of
   Amazon Linux 2023, Java, and other dependencies. Also fixed the scripts to be
   compatible to the recommended IMDSv2 authenticated APIs.
- Miscellaneous improvements and bug fixes across various sample templates.

---

By submitting this pull request, I confirm that you can use, modify, copy, and
redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: samples/sample-vpc/template.yml [hunk 1]
```diff
@@ -421,3 +391,146 @@ Resources:
...
+      Name:
+        Fn::Sub: ${Environment}-public-subnet-1c
+    Value:
+      Ref: PublicSubnet1c
+
+  VPC:
+    Description: The ID of the main VPC
+    Export:
+      Name:
+        Fn::Sub: ${Environment}-vpc-id
```
Comment (javydekoning) [inline samples/sample-vpc/template.yml:534]: Nit: we are mixing `!Sub` and `Fn::Sub` should we make it consistent? I sugges we use the short form (`!Sub`) everywhere as we are also using shortform for `!Ref` and `!GetAtt`

**Review by javydekoning — APPROVED** [review]

**Comment by javydekoning** [comment]
Note: linter is currently failing due to: https://github.com/tcort/markdown-link-check/issues/304 we can re-run or pin to @7.10.0

File: samples/sample-cdk-bootstrap/params/global.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+#     Default: ""
+#     Type: CommaDelimitedList
+#   FileAssetsBucketName:
+#     Description: The name of the S3 bucket used for file assets
+#     Default: ""
+#     Type: String
+#   FileAssetsBucketKmsKeyId:
+#     Description: Empty to create a new key (default), 'AWS_MANAGED_KEY' to use a managed S3 key, or the ID/ARN of an existing key.
+#     Default: ""
+#     Type: String
```
Comment (StewartW) [inline samples/sample-cdk-bootstrap/params/global.yml:24]: Why are these lines all commented out

Comment (sbkok) [reply samples/sample-cdk-bootstrap/params/global.yml:24]: To highlight what other parameters are available by the CDK Bootstrap stack (as generated by CDK).
Saves one looking it up by generating a template locally. Happy to remove it if you think that makes more sense.

**Review by StewartW — APPROVED** [review]

**Pull request merged**